python/lsst/obs/pfs/constructNormFlatTask.py

```diff
@@ -0,0 +1,205 @@
+            * scales      Scales to apply (expScales are scalings for each exposure,
+                               ccdScale is final scale for combined image)
+            * outputId    Data identifier for combined image (fully qualified for this CCD)
+        """
+        dataRefList = [getDataRef(cache.butler, dataId) if dataId is not None else None for
+                       dataId in struct.ccdIdList]
+        self.log.info("Combining %s on %s" % (struct.outputId, NODE))
+
+        self.log.info('len(dataRefList) = %d' % len(dataRefList))
+
+#        flatVisits = []#29,41,42,44,45,46,47,48,49,51,53]
+        
+        myFindTask = fataTask.FindAndTraceAperturesTask()
+        
+        exposure = dataRefList[0].get('postISRCCD')
+        
+        sumFlats = exposure.getMaskedImage().getImage().getArray()
+        sumVariances = exposure.getMaskedImage().getVariance().getArray()
+
+        allFts = []
+        xOffsets = []
+        for expRef in dataRefList:
+            exposure = expRef.get('postISRCCD')
+            xOffsets.append(exposure.getMetadata().get('sim.slit.xoffset'))
+            fts = myFindTask.run(exposure)
+            allFts.append(fts)
+            if expRef.dataId['visit'] != dataRefList[0].dataId['visit']:
+                sumFlats += exposure.getMaskedImage().getImage().getArray()
+                sumVariances += exposure.getMaskedImage().getVariance().getArray()
+
+        self.log.info('=== xOffsets = '+str(xOffsets)+' ===')
+
+        myProfileTask = cfftpTask.CreateFlatFiberTraceProfileTask()
+        myProfileTask.config.profileInterpolation = 'PISKUNOV'
+        myProfileTask.config.ccdReadOutNoise = 3.19
+        myProfileTask.config.maxIterSF = 15
+        myProfileTask.config.overSample = 15
+        myProfileTask.config.swathWidth = 250
+        myProfileTask.config.lambdaSF = 1./float(myProfileTask.config.overSample)
+        myProfileTask.config.lambdaSP = 0.
+        myProfileTask.config.wingSmoothFactor = 0.
+
+        for fts in allFts:
+            fts = myProfileTask.run(fts)
+
+        sumRec = np.ndarray(shape=sumFlats.shape, dtype='float32')
+        sumRec[:][:] = 0.
+        sumRecIm = afwImage.ImageF(sumRec)
+        rec = np.ndarray(shape=sumFlats.shape, dtype='float32')
+        rec[:][:] = 0.
+        recIm = afwImage.ImageF(rec)
+
+        sumVar = np.ndarray(shape=sumFlats.shape, dtype='float32')
+        sumVar[:][:] = 0.
+        sumVarIm = afwImage.ImageF(sumVar)
+
+        # Add all reconstructed FiberTraces of all dithered flats to one reconstructed image 'recIm'
+        for fts in allFts:
+            recIm.getArray()[:][:] = 0.
+            for ft in fts.getTraces():
+                spectrum = ft.extractFromProfile()
+                recFt = ft.getReconstructed2DSpectrum(spectrum)
+                recFtArr = recFt.getArray()
+                imArr = ft.getImage().getArray()
+                recFtArr = drpStella.where(imArr,'<=',0.,0.,recFtArr)
```

Comment:
What is this?  The C++ appears to do a string compare within an if block for every element of the array.